GPU/TPCFastTransformation/macro/TPCFastTransformInit.C

```diff
@@ -0,0 +1,159 @@
+///
+/// root -l TPCFastTransformInit.C'("debugVoxRes.root")'
+///
+
+#if !defined(__CLING__) || defined(__ROOTCLING__)
+
+#include "TFile.h"
+#include "TTree.h"
+#include "TSystem.h"
+
+#include "GPU/TPCFastTransform.h"
+#include "TPCReconstruction/TPCFastTransformHelperO2.h"
+#include "TPCSpaceCharge/SpaceCharge.h"
+#include "SpacePoints/TrackResiduals.h"
+
+#include "Riostream.h"
+
+#endif
+
+using namespace o2::tpc;
+using namespace o2::gpu;
+
+void TPCFastTransformInit(const char* fileName = "debugVoxRes.root")
+{
+  // open file with space-charge density histograms
+  // location: alien:///alice/cern.ch/user/e/ehellbar/TPCSpaceCharge/RUN3/ SCDensityHistograms/InputSCDensityHistograms.root
+  // The file contains several sc densities for epsilon=20 scenario
+  // Naming convention: inputSCDensity3D<nPileUpEvents>
+  //          - nPileUpEvents = 0,2,4,6,8,10,12
+  //                - number of events (in units of 1000) contributing to the ion density. 0 = 130000 events, 2 = 2000 events, 4 = 4000 events, etc.
+  //                - expected default map with 8000 pileup events: inputSCDensity3D8
+
+  if (gSystem->AccessPathName(fileName)) {
+    std::cout << " input file " << fileName << " does not exist!" << std::endl;
+    return;
+  }
+
+  auto file = std::unique_ptr<TFile>(TFile::Open(fileName, "READ"));
+  if (!file || !file->IsOpen()) {
+    std::cout << " input file " << fileName << " does not exist!" << std::endl;
+    return;
+  }
+
+  TTree* tree = nullptr;
+  file->cd();
+  gDirectory->GetObject("voxResTree", tree);
+  if (!tree) {
+    std::cout << "tree voxResTree does not exist!" << std::endl;
+    return;
+  }
+
+  o2::tpc::TrackResiduals::VoxRes* v;
+  tree->SetBranchAddress("voxRes", &v);
+
+  o2::tpc::TrackResiduals trackResiduals;
+  trackResiduals.init(); // also initializes the default binning which was used
+
+  auto* helper = o2::tpc::TPCFastTransformHelperO2::instance();
+  const o2::gpu::TPCFastTransformGeo& geo = helper->getGeometry();
+  o2::gpu::TPCFastSpaceChargeCorrectionMap& map = helper->getCorrectionMap();
+  map.init(geo.getNumberOfSlices(), geo.getNumberOfRows());
+
+  for (int iVox = 0; iVox < tree->GetEntriesFast(); iVox++) {
+    tree->GetEntry(iVox);
+    auto xBin = v->bvox[o2::tpc::TrackResiduals::VoxX];   // bin number in x (= pad row)
+    auto y2xBin = v->bvox[o2::tpc::TrackResiduals::VoxF]; // bin number in y/x 0..14
+    auto z2xBin = v->bvox[o2::tpc::TrackResiduals::VoxZ]; // bin number in z/x 0..4
+
+    auto x = trackResiduals.getX(xBin);             // radius of the pad row
+    auto y2x = trackResiduals.getY2X(xBin, y2xBin); // y/x coordinate of the bin ~-0.15 ... 0.15
+    auto z2x = trackResiduals.getZ2X(z2xBin);       // z/x coordinate of the bin 0.1 .. 0.9
+    float y = x * y2x;
+    float z = x * z2x;
+
+    float correctionX = v->D[o2::tpc::TrackResiduals::ResX];
+    float correctionY = v->D[o2::tpc::TrackResiduals::ResY];
+    float correctionZ = v->D[o2::tpc::TrackResiduals::ResZ];
```

Comment:
Eventually I think we would use the smoothed residuals as input (`v->DS`). But in the example which I sent you these are not filled. I also need to clean up the `VoxRes` struct, which I simply copied from AliRoot. E.g. the member `DC` for the Chebyshev parametrization will probably go. So for now should be absolutely fine to use `v->D` since its toy data only anyways.